src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerIntegrationTest.java

```diff
@@ -946,6 +946,29 @@ public void testHaplotypeCallerRemoveAltAlleleBasedOnHaptypeScores() throws IOEx
         }
     }
 
+    @Test(dataProvider="HaplotypeCallerTestInputs")
```

Comment:
I added some tests to check for strings instead of integers for ploidy values, and checking for negative values for ploidy. I'm not sure how to check against a reference mismatch since this supports bed files, which don't have any sort of sequence dictionary. Do you have any ideas? I suppose during construction I could check against the VCF header, but that only protects against differently named reference sequences, while the lengths could be different. (E.g. a hg38 <-> T2T mixup wouldn't be detectable with just a bed file)

Reply:
I think if the references mismatch subtly then we are hosed... but if somebody uses an HG19 bedfile + an HG38 reference this should at least warn the user because it should be easy to detect.... theoretically you could also check against the sequence dictionary that the intervals actually fall within the contig as well... Just asserting the names mismatch is probably strong enough protection